nmdc_automation/re_iding/scripts/re_id_tool.py

```diff
@@ -92,6 +91,242 @@ def cli(ctx, target):
+            )
+        result = db_client["metabolomics_analysis_activity_set"].delete_one(
+            {"id": orphan_metabolomics_activity_record["id"]}
+            )
+        logging.info(f"Deleted {result.deleted_count} metabolomics_analysis_activity_set records")
+
+
+def _update_metabolomics_input_data_objects(input_data_objects, db_client, api_client, updated_record_identifiers):
+    """
+    Update the input data objects for a Metabolomics Workflow Activity record.
+    """
+    new_input_data_object_ids = []
+    for input_data_object in input_data_objects:
+        # mint new ID for the data object
+        legacy_id = input_data_object["id"]
+        new_id = api_client.minter("nmdc:DataObject")
+        new_input_data_object_ids.append(new_id)
+        updated_record_identifiers.append(("data_object_set", legacy_id, new_id))
+        # update the data object record
+        do_filter_criteria = {"id": legacy_id}
+        do_update_criteria = {"$set": {"id": new_id, "alternative_identifiers": [legacy_id]}}
+        result = db_client["data_object_set"].update_one(do_filter_criteria, do_update_criteria)
+        logging.info(f"Updated {result.modified_count} data_object_set records")
+    return new_input_data_object_ids
+
+
+def _update_metabolomics_output_data_objects(output_data_objects, metabolomics_new_id, db_client, api_client,
+                                             updated_record_identifiers):
+    """
+    Update the output data objects for a Metabolomics OmicsProcessing record.
+    """
+    new_output_data_object_ids = []
+    for output_data_object in output_data_objects:
+        # mint new ID for the data object
+        legacy_id = output_data_object["id"]
+        new_id = api_client.minter("nmdc:DataObject")
+        new_output_data_object_ids.append(new_id)
+        updated_record_identifiers.append(("data_object_set", legacy_id, new_id))
+        # update the data object record
+        do_filter_criteria = {"id": legacy_id}
+        do_update_criteria = {"$set": {"id": new_id,  "was_generated_by": metabolomics_new_id}}
+        result = db_client["data_object_set"].update_one(do_filter_criteria, do_update_criteria)
+        logging.info(f"Updated {result.modified_count} data_object_set records")
+    return new_output_data_object_ids
+
+
+def _update_calibration_data_object(calibration_data_object, db_client, api_client, updated_record_identifiers):
+    """
+    Update the calibration data object for a Metabolomics Workflow Activity record.
+    """
+    # mint new ID for the data object
+    legacy_id = calibration_data_object["id"]
+    new_id = api_client.minter("nmdc:DataObject")
+    updated_record_identifiers.append(("data_object_set", legacy_id, new_id))
+    # update the data object record
+    do_filter_criteria = {"id": legacy_id}
+    do_update_criteria = {"$set": {"id": new_id, "alternative_identifiers": [legacy_id]}}
```

Comment:
remove updating alternative identifier for the calibration data_object_set records. This should only happen for data objects where are outputs of metabolomics or nom.

Reply:
https://github.com/microbiomedata/nmdc_automation/pull/116/commits/a30883e042258efc7c5a087a303788e397c95945